File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.kt
class Solution {
    fun solution(cards1: Array<String>, cards2: Array<String>, goal: Array<String>): String {
        var index1 = 0  // cards1 인덱스
        var index2 = 0  // cards2 인덱스

        for (word in goal) {
            when {
                index1 < cards1.size && cards1[index1] == word -> index1++
                index2 < cards2.size && cards2[index2] == word -> index2++
                else -> return "No"  // 둘 다 해당 안 되면 실패
            }
        }

        return "Yes"
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    i, j = 0, 0  # 각 카드덱의 포인터

    for word in goal:
        if i < len(cards1) and cards1[i] == word:
            i += 1
        elif j < len(cards2) and cards2[j] == word:
            j += 1
        else:
            return "No"  # 두 카드덱 어디에서도 현재 단어를 꺼낼 수 없으면 실패

    return "Yes"  # 전부 성공적으로 통과했다면
